File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalTest.java
package io.github.potjerodekool.openapi.internal.di.conditiontest;

import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition;

import java.lang.annotation.Annotation;

@FunctionalInterface
public interface ConditionalTest<C extends Annotation> {

    boolean test(C condition,
                 BeanDefinition beanDefinition);
}

File: code-generator/src/main/kotlin/io/github/potjerodekool/openapi/internal/di/conditiontest/ConditionalTest.kt
package io.github.potjerodekool.openapi.internal.di.conditiontest

import io.github.potjerodekool.openapi.internal.di.bean.BeanDefinition

fun interface ConditionalTest<C : Annotation> {
    fun test(
        condition: C,
        beanDefinition: BeanDefinition
    ): Boolean
}
